src/native/tools/kube_pv_chown.cpp

```diff
@@ -35,6 +37,28 @@ change_path_permissions(const char* path, long uid)
     return 0;
 }
 
+void
+change_path_permissions_recursive(std::string path, long uid)
+{
+    change_path_permissions(path.c_str(), uid);
+    DIR* dir = opendir(path.c_str());
+    if (!dir) {
+        if (errno != ENOTDIR) {
+            cerr << "Error: opendir(" << path << ") failed " << strerror(errno) << endl;
+            exit(1);
+        }
+        return;
+    }
+    struct dirent* entry;
+    while ((entry = readdir(dir)) != NULL) {
+        if (std::string(entry->d_name).compare(".") != 0 && std::string(entry->d_name).compare("..") != 0) {
```

Comment:
I think this can be more readable:
```suggestion
        std::string entry_name = entry->d_name;
        std::string entry_path = path + "/" + entry_name;
        if (entry_name != "." && entry_name != "..") {
            change_path_permissions_recursive(entry_path, uid);
        }
```